src/distributions/uniform.rs

```diff
@@ -847,13 +980,95 @@ mod tests {
                             assert!(low <= v && v < high);
                         }
                     }
-                 )*
+                )*
             }}
         }
         t!(i8, i16, i32, i64, isize,
            u8, u16, u32, u64, usize);
         #[cfg(feature = "i128_support")]
-        t!(i128, u128)
+        t!(i128, u128);
+
+        #[cfg(feature = "simd_support")]
+        {
+            macro_rules! t_simd {
+                ($($ty:ident),* => $elem_ty:ident) => {{
+                    $(
+                        let v: &[($ty, $ty)] = &[
+                            ($ty::splat(0), $ty::splat(10)),
+                            ($ty::splat(10), $ty::splat(127)),
```

Comment:
Could you fold this into the existing macro by using `FloatSIMDUtils::cast_from_int` and `FloatSIMDUtils::all_le`?

Or rather, replace the existing macro with your new macro, and just add the scalar types to it.

I guess we'd still have to keep the full-range tests separate unless we make the changes above.